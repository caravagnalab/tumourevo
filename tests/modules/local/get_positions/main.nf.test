nextflow_process {

    name "Test Process GET_POSITIONS_ALL"
    script "modules/local/get_positions/main.nf"
    process "GET_POSITIONS_ALL"

    test("Should run without failures") {

        when {
            params {
                // define module specific parameters here
                // outdir = "tests/results"
            }
            process {
                """
                meta = [
                    id : "MSeq_Set06_Set6_42",
                    dataset : "MSeq",
                    patient : "Set06",
                    tumour_sample: "Set6_42",
                    normal_sample: "54_N",
                    cancer_type: "PANCANCER" 
                ]
                rds_list = file(params.local_testdata_base_path + "/from_vcf2CNAqc/MSeq_Set06_Set6_42_snv.rds", checkIfExists: true)
                sample_list = "Set6_42"
                input[0] = tuple(meta, rds_list)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
