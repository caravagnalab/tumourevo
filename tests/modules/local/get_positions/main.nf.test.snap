{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "MSeq_Set06_Set6_42",
                            "dataset": "MSeq",
                            "patient": "Set06",
                            "tumour_sample": "Set6_42",
                            "normal_sample": "54_N",
                            "cancer_type": "PANCANCER"
                        },
                        "MSeq_Set06_Set6_42_all_positions.rds:md5,af193f793262106018e3d946bf36fc69"
                    ]
                ],
                "all_pos": [
                    [
                        {
                            "id": "MSeq_Set06_Set6_42",
                            "dataset": "MSeq",
                            "patient": "Set06",
                            "tumour_sample": "Set6_42",
                            "normal_sample": "54_N",
                            "cancer_type": "PANCANCER"
                        },
                        "MSeq_Set06_Set6_42_all_positions.rds:md5,af193f793262106018e3d946bf36fc69"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-09-16T12:27:04.35969047"
    }
}